py/src/nupic/encoders/grid_cell_encoder.py

```diff
@@ -0,0 +1,136 @@
+# ------------------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+#
+# Copyright (C) 2018-2019, David McDougall
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero Public License version 3 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License along with
+# this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ------------------------------------------------------------------------------
+
+
+import numpy as np
+import math
+from nupic.encoders import hexy
+from nupic.bindings.sdr import SDR
+from nupic.bindings.math import Random
+
+class GridCellEncoder:
+    """ TODO: DOCUMENTATION """
+    def __init__(self,
+            size,
+            sparsity,
+            periods,
+            seed = 0,):
+        """ TODO: DOCUMENTATION """
+        self.size       = size
+        self.dimensions = (size,)
+        self.sparsity   = sparsity
+        self.periods    = tuple(sorted(periods))
+        assert(min(self.periods) >= 4)
+
+        # Assign each module a range of cells in the output SDR.
+        partitions       = np.linspace(0, self.size, num=len(self.periods) + 1)
+        partitions       = list(zip(partitions, partitions[1:]))
+        self.partitions_ = [(int(round(start)), int(round(stop)))
+                                        for start, stop in partitions]
+
+        # Assign each module a random offset and orientation.
+        rng            = np.random.RandomState(seed = Random(seed).getUInt32())
```

Comment:
why not use our cpp-binded Random?

Reply:
I used numpy.random so that I could use `rng.uniform()` which returns a numpy array.  The alternative would be to make the numpy array and iterate through it to fill in the values.

TODO: Add a test which verifies that the GridCellEncoder seed is deterministic & cross platform.